app/src/features/apiClient/screens/apiClient/components/sidebar/components/collectionsList/collectionRow/CollectionRow.tsx

```diff
@@ -70,17 +79,30 @@ export const CollectionRow: React.FC<Props> = ({ record, onNewClick, onExportCli
     [setIsDeleteModalOpen, updateRecordToBeDeleted]
   );
 
-  const collapseChangeHandler = useCallback((keys: string[]) => {
-    if (isEmpty(keys)) {
-      collapsedKeys.push(record.id);
-    } else {
-      const activeKeyIndex = collapsedKeys.indexOf(record.id);
-      collapsedKeys.splice(activeKeyIndex, 1);
-    }
-    setActiveKey(keys[0]);
-    isEmpty(collapsedKeys)
-      ? sessionStorage.removeItem("collapsed_collection_keys")
-      : sessionStorage.setItem("collapsed_collection_keys", collapsedKeys);
+  const collapseChangeHandler = useCallback(
+    (keys: RQAPI.Record["id"][]) => {
+      let activeKeysCopy = [...activeKeys];
+      if (isEmpty(keys)) {
+        activeKeysCopy = activeKeysCopy.filter((key) => key !== record.id);
+      } else if (!activeKeysCopy.includes(record.id)) {
+        activeKeysCopy.push(record.id);
+      }
+      updateActiveKeys(activeKeysCopy);
+      isEmpty(activeKeysCopy)
+        ? sessionStorage.removeItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY)
+        : sessionStorage.setItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, activeKeysCopy);
+    },
+    [record, activeKeys]
+  );
+
+  useEffect(() => {
+    setActiveKey(activeKeys?.includes(record.id) ? record.id : null);
+  }, [activeKeys, record.id]);
```

Comment:
No need of state and useEffect
```suggestion
 const activeKey = activeKeys?.includes(record.id) ? record.id : null
```